Sources/SearchResults.swift

```diff
@@ -428,6 +453,30 @@ private func swift2Objc(_ matchLevel: MatchLevel_?) -> MatchLevel {
         return returnedFacets[name] as? [String: Int]
     }
     
+    /// Retrieve the hierarchical facet values for a given facet.
+    ///
+    /// - parameter name: Facet name.
+    /// - returns: The corresponding hierarchical facet values (which may be empty).
+    ///
+    @objc public func hierarchicalFacetValues(name: String) -> [HierarchicalFacet] {
+        let separator = " > "
+        guard let attributes = hierarchicalFacets[name], !attributes.isEmpty else { return [] }
+        let values = attributes.map { facets(name: $0) ?? [:] }
+        let rootLevel = values.first?.map { HierarchicalFacet(value: $0.key, count: $0.value) } ?? []
+        if values.count > 1 {
```

Comment:
To avoid indentation, you can do `guard values.count > 1 else { return rootLevel }`